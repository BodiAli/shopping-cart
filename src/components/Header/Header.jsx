import { NavLink } from "react-router-dom";
import styles from "./Header.module.css";

function Header() {
  return (
    <>
      <header className={styles.header}>
        <h1>GameVault</h1>
        <div className={styles.navContainer}>
          <NavLink to="/">Home</NavLink>
          <NavLink to="shop">Shop</NavLink>
          <NavLink to="about">About</NavLink>
        </div>
      </header>
    </>
  );
}

export default Header;
