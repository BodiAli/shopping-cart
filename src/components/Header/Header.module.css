.header {
  background-color: var(--primary);
  padding: 1.5rem 1rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.header > h1 {
  font-family: "Merriweather", Times, serif;
  font-size: 2.2rem;
  font-weight: bold;
  background-image: url(/green-bg-text.jpg);
  background-clip: text;
  background-position: center;
  filter: brightness(1.8);
  color: transparent;
}

.navContainer {
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 100px;
}

.navContainer > a {
  text-decoration: none;
  color: var(--darkerGreen);
  padding: 0.6rem 1.2rem;
  border-radius: 5px;
  background-color: var(--darkerGray);
}
