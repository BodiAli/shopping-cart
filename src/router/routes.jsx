import App from "../App";
import ErrorPage from "../pages/ErrorPage/ErrorPage";
import HomePage from "../pages/HomePage/HomePage";
import ErrorComponent from "../components/ErrorComponent/ErrorComponent";
import ShopPage from "../pages/ShopPage/ShopPage";
import Category from "../components/Category/Category";
import ProductPage from "../pages/ProductPage/ProductPage";
import AboutPage from "../pages/AboutPage/AboutPage";

const routes = [
  {
    path: "/",
    element: <App />,
    errorElement: <ErrorPage />,
    children: [
      {
        errorElement: <ErrorComponent />,
        children: [
          {
            index: true,
            element: <HomePage />,
            loader: HomePage.loader,
          },
          {
            path: "shop",
            element: <ShopPage />,
            children: [
              {
                path: ":category",
                element: <Category />,
                loader: Category.loader,
                errorElement: <ErrorComponent />,
              },
            ],
          },
          {
            path: "product/:id",
            element: <ProductPage />,
            loader: ProductPage.loader,
          },
          {
            path: "about",
            element: <AboutPage />,
          },
        ],
      },
    ],
  },
];

export default routes;
