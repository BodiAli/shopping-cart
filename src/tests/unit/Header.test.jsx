import { render, screen } from "@testing-library/react";
import Header from "../../components/Header/Header";
import { createMemoryRouter, RouterProvider } from "react-router-dom";
import routes from "../../router/routes";
import { expect } from "vitest";

const router = createMemoryRouter(routes);
const wrapper = () => <RouterProvider router={router}></RouterProvider>;

describe("Header component", () => {
  test("should render correct elements", () => {
    const { container } = render(<Header />, { wrapper: wrapper });
    expect(container).toMatchSnapshot();
  });

  test("should render header element", () => {
    render(<Header />, { wrapper: wrapper });
    expect(screen.getByRole("heading")).toBeInTheDocument();
  });

  test("should render nav buttons", () => {
    render(<Header />, { wrapper: wrapper });
    expect(screen.getByRole("link", { name: /home/i })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /shop/i })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /about/i })).toBeInTheDocument();
  });
});
