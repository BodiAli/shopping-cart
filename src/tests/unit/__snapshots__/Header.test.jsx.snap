// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > should render correct elements 1`] = `
<div>
  <header
    class="_header_63c2ea"
  >
    <h1>
      GameVault
    </h1>
    <div
      class="_navContainer_63c2ea"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Home
      </a>
      <a
        class=""
        href="/shop"
      >
        Shop
      </a>
      <a
        class=""
        href="/about"
      >
        About
      </a>
    </div>
  </header>
</div>
`;
